SCons/Tool/applelink.py

```diff
@@ -71,12 +68,7 @@ def _applelib_versioned_lib_soname(env, libnode, version, prefix, suffix, name_f
         print("_applelib_versioned_lib_soname: name={!r}".format(name))
     major = version.split('.')[0]
     (libname,_suffix) = name.split('.')
-    # if a desired SONAME was supplied, use that, otherwise create 
```

Comment:
Why remove this?

Reply:
This was a fix to #3246, but it was not the correct way to fix it, because ShLibSonameGenerator is meant to call the function it was called in.

Reply:
I'm not sure what you mean by "call the function it was called in". Can you elaborate?

Reply:
Sorry, when I responded to this I got ripped away from the computer before I could write it all out. 

The ShLibSonameGenerator (really a specific instance of the base class _LibSonameGenerator for shlibs) is the base generator of sonames. If you look at the class, when there is no SONAME defined, the _LibSonameGenerator will attempt to call the generate_versioned_lib_info, which then gets and calls a specific callback from env['LINKCALLBACKS'], which in in applelink would be this function.

The real issue from #3246 was in ```_versioned_shlib_symlinks``` (updated in this PR) which should be setting soname_func as ShLibSonameGenerator, was just skipping down to the LINKCALLBACK for the specific type of library. That function should have let ShLibSonameGenerator decide to call the LINKCALLBACK if the SONAME was not defined, instead of the specific library function making the decision.